mailbox/api/src/main/java/org/apache/james/mailbox/model/FetchGroupImpl.java

```diff
@@ -29,38 +31,42 @@
  */
 public class FetchGroupImpl implements MessageResult.FetchGroup {
 
-    public static final MessageResult.FetchGroup MINIMAL = new FetchGroupImpl(MessageResult.FetchGroup.MINIMAL);
+    public static final MessageResult.FetchGroup MINIMAL = new FetchGroupImpl(EnumSet.noneOf(MessageResult.FetchGroupEnum.class));
```

Comment:
IMHO it is risky to have mutable global constant. Can the FetchGroupImpl be immutable ? Or is it planned in another PR ?

Reply:
I did immutability related changes in https://github.com/linagora/james-project/pull/2928

I am going to backport @mbaechler refactoring on top of this changeset (once merged)

For the record, I did screw up my FetchGroupTest because of these mutable shared contant (hence the **strong** desire to get this immutable).